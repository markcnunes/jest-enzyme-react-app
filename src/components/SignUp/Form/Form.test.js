import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import { act } from "react-dom/test-utils";

jest.mock("./FavouriteAuthor", () => {
  const React = require("react");

  return {
    __esModule: true,
    default: React.forwardRef((props, ref) => {
      return (
        <select name="favouriteAuthor" ref={ref}>
          <option value="test-author">Test Author</option>
        </select>
      );
    }),
  };
});

test("submits correctly", async () => {
  const handleCompleteMock = jest.fn();
  const signUpAPIMock = jest.fn();

  jest.doMock("./_signUp", () => ({
    __esModule: true,
    default: signUpAPIMock,
  }));

  const Form = require("./Form").default;

  let wrapper;

  await act(async () => {
    wrapper = mount(<Form onComplete={handleCompleteMock} />);
    wrapper.find("input[name='fullName']").instance().value = "John";
    wrapper.find("input[name='email']").instance().value = "john@example.com";
    wrapper.find("form").simulate("submit");
  });

  expect(signUpAPIMock).toHaveBeenCalledWith({
    email: "john@example.com",
    favouriteAuthor: "test-author",
    fullName: "John",
  });
  expect(handleCompleteMock).toHaveBeenCalled();
});

test("matches snapshot", async () => {
  const Form = require("./Form").default;

  let wrapper;

  await act(async () => {
    wrapper = mount(<Form />);
  });

  expect(toJSON(wrapper)).toMatchSnapshot();
});
