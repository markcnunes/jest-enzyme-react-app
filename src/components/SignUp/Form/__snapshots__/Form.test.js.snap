// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Form>
  <form
    onSubmit={[Function]}
  >
    <styled.label>
      <label
        className="sc-AxhCb QqhkI"
      >
        <styled.span>
          <span
            className="sc-AxhUy kNHWOG"
          >
            Full Name
          </span>
        </styled.span>
        <styled.input
          autoFocus={true}
          disabled={false}
          name="fullName"
          type="text"
        >
          <input
            autoFocus={true}
            className="sc-AxiKw dRwlfD"
            disabled={false}
            name="fullName"
            type="text"
          />
        </styled.input>
      </label>
    </styled.label>
    <styled.label>
      <label
        className="sc-AxhCb QqhkI"
      >
        <styled.span>
          <span
            className="sc-AxhUy kNHWOG"
          >
            Email Address
          </span>
        </styled.span>
        <styled.input
          disabled={false}
          name="email"
          type="email"
        >
          <input
            className="sc-AxiKw dRwlfD"
            disabled={false}
            name="email"
            type="email"
          />
        </styled.input>
      </label>
    </styled.label>
    <styled.label>
      <label
        className="sc-AxhCb QqhkI"
      >
        <styled.span>
          <span
            className="sc-AxhUy kNHWOG"
          >
            Favourite Author
          </span>
        </styled.span>
        <ForwardRef
          disabled={false}
        >
          <select
            name="favouriteAuthor"
          >
            <option
              value="test-author"
            >
              Test Author
            </option>
          </select>
        </ForwardRef>
      </label>
    </styled.label>
    <styled.div>
      <div
        className="sc-AxirZ gNUgBZ"
      >
        <styled.button
          disabled={false}
          type="submit"
        >
          <button
            className="sc-AxjAm hriFsm"
            disabled={false}
            type="submit"
          >
            Sign Up
          </button>
        </styled.button>
      </div>
    </styled.div>
  </form>
</Form>
`;
